ext/bg/js/search.js

```diff
@@ -271,19 +268,24 @@ class DisplaySearch extends Display {
         }
     }
 
-    async updateOptions(options) {
-        await super.updateOptions(options);
+    onProfileChanged(profileIndex) {
+        super.onProfileChanged(profileIndex);
+        const query = this.query.value;
+        if (query) {
+            this.setQuery(query);
+            this.onSearchQueryUpdated(query, false);
+        }
+    }
```

Comment:
This will be called from `prepare` before `QueryParser` has been prepared. It throws an error about missing template handler. Will have to look for a workaround, for example using a prepared flag.